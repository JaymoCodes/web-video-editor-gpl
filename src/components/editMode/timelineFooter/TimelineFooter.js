/* 
 *  TimelineFooter component: in this component we display video's final time output
 */

import React, { Component } from 'react';
import styles from './TimelineFooter.module.css';
import PropTypes from 'prop-types';

import { sec2duration } from '../../../utils/time';

class TimelineFooter extends Component {

  /*
   * Compute the length and if it's >=0 display it 
   * If it's not, then startTrim and/or endTrim are not valid/inserted so display an error message
   */ 

  render() {
    const length = this.props.endTrim - this.props.startTrim;

    if (length > 0)
      return (
        <div className={styles.output}>
          Final Output &nbsp; <span>{sec2duration(length)}</span>
        </div>
      );

    return (
      <div className={styles.output}>
        Please select valid start and end time
      </div>
    );
  }
}

TimelineFooter.propTypes = {
  startTrim: PropTypes.number,
  endTrim: PropTypes.number,
};

export default TimelineFooter;
