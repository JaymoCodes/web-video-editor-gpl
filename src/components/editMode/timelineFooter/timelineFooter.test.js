import React from 'react';
import renderer from 'react-test-renderer';
import TimelineFooter from './TimelineFooter';

describe('TimelineFooter', () => {
  it('Renders without crashing', () => {
    let wrapper, tree;
  
    wrapper = renderer.create(
      <TimelineFooter />
    );
  
    tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    wrapper = renderer.create(
      <TimelineFooter
        startTrim={10.0}
        endTrim={15.2} 
      />
    );

    tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

  });
});
