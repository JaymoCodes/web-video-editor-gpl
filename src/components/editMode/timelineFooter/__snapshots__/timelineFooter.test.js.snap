// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineFooter Renders without crashing 1`] = `
<div
  className="output"
>
  Please select valid start and end time
</div>
`;

exports[`TimelineFooter Renders without crashing 2`] = `
<div
  className="output"
>
  Final Output Â  
  <span>
    05.20
  </span>
</div>
`;
